src/java.base/share/classes/jdk/internal/util/access/ByteArrayAccess.java

```diff
@@ -0,0 +1,790 @@
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using big endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws NullPointerException      if the provided {@code array} is {@code null}
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 4]
+     * @see #getLong(byte[], int)
+     */
+    public static void setLong(byte[] array, int offset, long value) {
+        LONG.set(array, offset, value);
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using big endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * Variants of {@linkplain Double#NaN } values are canonized to a single NaN value.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws NullPointerException      if the provided {@code array} is {@code null}
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 2]
+     * @see #getDouble(byte[], int)
+     */
+    public static void setDouble(byte[] array, int offset, double value) {
+        // Using Double.doubleToLongBits collapses NaN values to a single
+        // "canonical" NaN value
+        LONG.set(array, offset, Double.doubleToLongBits(value));
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using big endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * Variants of {@linkplain Double#NaN } values are silently written according to
+     * their bit patterns.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws NullPointerException      if the provided {@code array} is {@code null}
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 2]
+     * @see #getDoubleRaw(byte[], int)
+     */
+    public static void setDoubleRaw(byte[] array, int offset, double value) {
+        // Just sets the bits as they are
+        DOUBLE.set(array, offset, value);
+    }
+
+
+    /*
+     * Methods for packing primitive values into byte arrays starting at offset zero.
+     */
```

Comment:
Is the only advantage to the zero offset versions in the API. 
Is there any performance difference?

Reply:
Very good question. I will take a look at it.

Reply:
No significant performance increase:

```
2-parameters
Benchmark                                             Mode  Cnt  Score   Error  Units
PrimitiveFieldSerializationBenchmark.serializeData    avgt    8  6.761 ± 0.126  ns/op
PrimitiveFieldSerializationBenchmark.serializeRecord  avgt    8  6.890 ± 0.093  ns/op

3-parameters
Benchmark                                             Mode  Cnt  Score   Error  Units
PrimitiveFieldSerializationBenchmark.serializeData    avgt    8  6.850 ± 0.074  ns/op
PrimitiveFieldSerializationBenchmark.serializeRecord  avgt    8  6.855 ± 0.057  ns/op
```

Reply:
I will remove the extra methods with lesser parameters.